lms/data_tasks/report/create_from_scratch/01_functions/01_date_functions.sql

```diff
@@ -11,7 +11,15 @@ IMMUTABLE
 AS $$
     SELECT CASE
         WHEN resolution = 'all_time' THEN '1901-01-01'::DATE
-        WHEN resolution = 'academic_year' THEN (DATE_TRUNC('year', date - INTERVAL '6 month') + INTERVAL '6 month')::DATE
+        WHEN resolution = 'academic_year' THEN (
+            DATE_TRUNC('year', date - INTERVAL '6 month') + INTERVAL '6 month'
+        )::DATE
+        -- Trailing year is a time period based on "today" as the end-period
+        -- quantized by year. We get this by truncating to the year, a year ago
+        -- and then adding the portion through the year we `NOW()` back on.
+        WHEN resolution = 'trailing_year' THEN (
+            DATE_TRUNC('year', date - INTERVAL '1 year')  + NOW() - DATE_TRUNC('year', NOW())
```

Comment:
I can't help but think there's a simplification we can here, but I can't spot it.